### DISCUSSION TITLE
compilation error : 'INTEGER_ATTRIBUTE' has no member named 'nominal'

### DISCUSSION THREAD

**Discussion by denis-jidovtseff-jc** [OpenModelica/OpenModelica#10244] [General]
when I'm running a model I have a compilation issue

```
StationDynHPLP_02nls.c:4044:122: error: 'INTEGER_ATTRIBUTE' has no member named 'nominal'
 4044 |   sysData->nominal[i] = data->modelData->integerVarsData[66].attribute /* compressor.mediumA.phase */.nominal;

```
I can't give more information on the code because of IP.

Does someone can explain me when this error can occur? what's could be the reason?

**Comment by sjoelund** [comment]
The backend decided to try to solve an nonlinear system involving integers. This is not really possible to solve for us and should ideally be detected somewhere in the compiler backend telling you which variables are involved and why we cannot solve a system involving discrete variables at that place...

**Comment by mahge** [comment]
Until @casella gets here and give you some more detailed information, you can try the flag `--tearingMethod=minimalTearing`. This will try to avoid picking discrete variables as iteration variables for solving non-linear systems.

**Comment by casella** [comment]
I'm actually not the real expert here, @phannebohm and @kabdelhak are. The issue is described in #9921, where it is [explicitly stated](https://github.com/OpenModelica/OpenModelica/issues/9921#issuecomment-1347566061) that our current nonlinear solvers are not meant to involve discrete variables as iteration variables. 

As a consequence, implicit systems of equations involving discrete (Boolean or Integer) variables should always be solved by tearing, where the discrete variables are part of the torn (or inner) variables, that are computed explicitly from the (continuous) iteration variables. 

You get that behaviour if you select --tearingMethod=minimalTearing, but that will disable all other useful tearing of continuous variables. Unfortunately it is difficult to modify the tearing algorithms so that the guarantee to tear as many continuous variables as possible and all the discrete variables at the same time, so sometimes discrete variables end up in the set of iteration variables. One option I suggested in #9921 is to try out regular tearing and if some discrete variables end up being picked as iteration variables, discard it and use minimal tearing instead, automatically.

This should be on @phannebohm's todo list.

**Reply by denis-jidovtseff-jc** [reply]
thanks, it solve the compilation problem
